kombu/utils/url.py

```diff
@@ -120,3 +118,20 @@ def maybe_sanitize_url(url, mask='**'):
     if isinstance(url, string_t) and '://' in url:
         return sanitize_url(url, mask)
     return url
+
+
+def parse_ssl_cert_reqs(query_value):
+    # type: (str) -> Any
+    """Given the query parameter for ssl_cert_reqs, return the SSL constant or None."""
+    if ssl_available:
+        query_value_to_constant = {
```

Comment:
You can specify this mapping as a constant, e.g. `SSL_CERT_REQS`. Although there is minimal performance benefit, as this code runs only initialization, a new dictionary will not have to be created each time.

Reply:
Well this is being called within a loop so if the method were reached for multiple keys then it would generate multiple times on initalization.  However upon closer inspection above, it only gets into this section if key is key == 'ssl_cert_reqs' and so then then that is the only key that would ever operate on `query[key] = parse_ssl_cert_reqs(query[key])` ==>  `query['ssl_cert_reqs'] = parse_ssl_cert_reqs(query['ssl_cert_reqs'])`.

So then query of ssl_cert_reqs will only ever be one of the dictionary values per iteration-- this is why I know generating a dictionary each time is not ideal because we are doing this in some kind if iterative calling pattern.   Plus any part of the eventloop that restarts or reinitialize would happen again.  

Anyway, Happy Holidays!!!

Reply:
Happy to refactor, but I'm skeptical of the benefit here. Using a constant means the dict will stick around in memory after initialization even though it's never going to be needed again. The analogous PR for `celery` (https://github.com/celery/celery/pull/5703) also uses an inline dict.

Reply:
@jdavisp3 you are right! However, in terms of semantics this is a constant mapping, so this is what I meant and I should probably have explained it better, sorry for that. I thought that by converting to a constant, there are multiple gains (minor performance gain and slightly more understandable code for future readers). Either way this is not so important, but since this is a quite large and complex codebase, it is sometimes better to pay closer attention to these details.